hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/HealthyPipelineSafeModeRule.java

```diff
@@ -193,9 +200,20 @@ public synchronized int getHealthyPipelineThresholdCount() {
 
   @Override
   public String getStatusText() {
-    return String.format("healthy Ratis/THREE pipelines (=%d) >= "
-            + "healthyPipelineThresholdCount (=%d)",
-        getCurrentHealthyPipelineCount(),
+    String status = String.format(
+        "healthy Ratis/THREE pipelines (=%d) >= healthyPipelineThresholdCount" +
+            " (=%d)", getCurrentHealthyPipelineCount(),
         getHealthyPipelineThresholdCount());
+    Set<PipelineID> samplePipelines = pipelineIDSet.stream()
+        .filter(element -> !processedPipelineIDs.contains(element))
+        .limit(SAMPLE_PIPELINE_DISPLAY_LIMIT)
+        .collect(Collectors.toSet());
```

Comment:
Instead of filtering here, we could remove pipeline IDs as they are being processed:

https://github.com/apache/ozone/blob/a05bfb9ad53ca745e75362729993b3f6083d61f1/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/safemode/HealthyPipelineSafeModeRule.java#L130-L137

Reply:
Done.